launcher/minecraft/launch/VerifyJavaInstall.cpp

```diff
@@ -39,6 +39,15 @@
 #include "minecraft/PackProfile.h"
 #include "minecraft/MinecraftInstance.h"
 
+#ifdef Q_OS_FREEBSD
+    #ifdef major
+        #undef major
+    #endif
+    #ifdef minor
+        #undef minor
+    #endif
```

Comment:
You shouldn't need to `ifdef` for every macro. Just `undef` should do.

Reply:
Also, the ifdef check for FreeBSD shouldn't be necessary. Any definition of macros with those names is likely to break things, it doesn't really need to only undef them specifically on FreeBSD

Reply:
So here I go